internal/btf/strings.go

```diff
@@ -50,5 +51,10 @@ func (st stringTable) Lookup(offset uint32) (string, error) {
 		return "", fmt.Errorf("offset %d isn't null terminated", offset)
 	}
 
-	return string(str[:end]), nil
+	return unsafeBytesToString(str[:end]), nil
+}
+
+func unsafeBytesToString(bytes []byte) string {
+	// trick used by https://go.dev/src/strings/builder.go#L45
+	return *(*string)(unsafe.Pointer(&bytes))
```

Comment:
maybe a personal preference:
I prefer the use of `reflect.StringHeader{}` over this approach. If `bytes` contains malicious data and `Len` in `reflect.StringHeader{}` is not set, it can result in unintended side effects when using the string.
Both approaches avoid additional allocations so this is an improvement memory wise.

Reply:
I gave your suggestion a quick whirl:

```Go
func unsafeBytesToString(bytes []byte) string {
	if len(bytes) == 0 {
		return ""
	}

	sh := &reflect.StringHeader{
		Data: uintptr(unsafe.Pointer(&bytes[0])),
		Len:  len(bytes),
	}

	str := *(*string)(unsafe.Pointer(sh))
	runtime.KeepAlive(bytes)
	return str
}
```

Seems to work, though some linter complains about this being a possible misuse of StringHeader. What do you mean with `bytes` containing malicious data though?

Reply:
Another thing I'm trying to figure out: this trick effectively pins the entire string table in memory. So while we allocate less overall, the peak memory usage after parsing might be higher. I'm thinking of a case where a user parses vmlinux, gets out some type they are interested in, and afterwards discards the Spec. The single retained type would lock the string table into memory.

Reply:
This is true sadly, and it may not be acceptable